gov2/s3/CopyObject/CopyObjectv2.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+// snippet-start:[s3.go-v2.CopyObject]
+package main
+
+// snippet-start:[s3.go-v2.CopyObject.imports]
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+// snippet-end:[s3.go-v2.CopyObject.imports]
+
+// S3CopyObjectAPI defines the interface for the Amazon S3 CopyObject function.
```

Comment:
"Amazon Simple Storage Service (Amazon S3)" as close as possible to first mention in each code file, here and throughout.